receiver/snmpreceiver/factory.go

```diff
@@ -69,3 +79,39 @@ func createMetricsReceiver(
 
 	return scraperhelper.NewScraperControllerReceiver(&snmpConfig.ScraperControllerSettings, params, consumer, scraperhelper.AddScraper(scraper))
 }
+
+// addMissingConfigDefaults adds any missing comfig parameters that have defaults
+func addMissingConfigDefaults(cfg *Config) {
```

Comment:
Really this should be done in a custom unmarshaler. The reason is that the config has already been validated, but now we're changing it. That said, I think it's probably ok as long as we call `.Validate` manually here. We should plan on moving it to a custom unmarshaler later though.